### ISSUE TITLE
php7 branch

### ISSUE THREAD

**Issue by pjebs** [php-memcached-dev/php-memcached#194]
I ran this code below:

`
git clone https://github.com/php-memcached-dev/php-memcached

cd php-memcached

git checkout php7

phpize

./configure

make

sudo make install
`

It didn't work. The response is below:

make
make: *** No targets specified and no makefile found.  Stop.

**Comment by isotopp** [comment]
Is the phpize command you are calling a phpize from PHP 7? Because if you checkout the php7 branch of php-memcached, the phpize must match.

(The sequence does work for me with a self-build php7, a php7 phpize from the same build and as shown by you)

**Comment by isotopp** [comment]
{code bash}
[kris@kk2:~/Source/php-memcached] php7* ± which phpize
/opt/php/bin/phpize

[kris@kk2:~/Source/php-memcached] php7* ± which php
/opt/php/bin/php

[kris@kk2:~/Source/php-memcached] master* ± php -v
PHP 7.1.0-dev (cli) (built: Nov  9 2015 20:19:56) ( NTS )
Copyright (c) 1997-2015 The PHP Group
Zend Engine v3.1.0-dev, Copyright (c) 1998-2015 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2015, by Zend Technologies

[kris@kk2:~/Source/php-memcached] master* ± git checkout php7
Switched to branch 'php7'
Your branch is up-to-date with 'origin/php7'.

[kris@kk2:~/Source/php-memcached] php7* ± phpize
Configuring for:
PHP Api Version:         20151012
Zend Module Api No:      20151012
Zend Extension Api No:   320151012

[kris@kk2:~/Source/php-memcached] php7* ± ./configure --with-zlib-dir=/usr/local/Cellar/zlib/1.2.8/
...
creating libtool
appending configuration tag "CXX" to libtool
configure: creating ./config.status
config.status: creating config.h
config.status: config.h is unchanged

[kris@kk2:~/Source/php-memcached] php7* 3s ± make -j 4
...
cp ./.libs/memcached.so /Users/kris/Source/php-memcached/modules/memcached.so
cp ./.libs/memcached.lai /Users/kris/Source/php-memcached/modules/memcached.la
----------------------------------------------------------------------
Libraries have been installed in:
   /Users/kris/Source/php-memcached/modules

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the `-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the `DYLD_LIBRARY_PATH' environment variable
     during execution

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------

Build complete.
Don't forget to run 'make test'.

[kris@kk2:~/Source/php-memcached] php7* ± make install
Installing shared extensions:     /opt/php/lib/php/extensions/no-debug-non-zts-20151012/
{code}

**Comment by jonathanpmartins** [comment]
I think my compilation is not right too. I can compile the php7 branch, but the ```make test``` command fails. The list is very long!
When I run ```make install``` it seems that the extension is installed correctly.

    php -v

```
PHP 7.0.2-4+deb.sury.org~trusty+1 (cli) ( NTS )
Copyright (c) 1997-2015 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2015 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2015, by Zend Technologies
```

    php -i | grep memcached

```
memcached
memcached support => enabled
libmemcached version => 1.0.18
memcached.compression_factor => 1.3 => 1.3
memcached.compression_threshold => 2000 => 2000
memcached.compression_type => fastlz => fastlz
memcached.serializer => php => php
memcached.sess_binary_protocol => 1 => 1
memcached.sess_compression => 1 => 1
memcached.sess_connect_timeout => 0 => 0
memcached.sess_consistent_hash => 1 => 1
memcached.sess_lock_expire => 0 => 0
memcached.sess_lock_max_wait => not set => not set
memcached.sess_lock_retries => 5 => 5
memcached.sess_lock_wait => not set => not set
memcached.sess_lock_wait_max => 2000 => 2000
memcached.sess_lock_wait_min => 1000 => 1000
memcached.sess_locking => 1 => 1
memcached.sess_number_of_replicas => 0 => 0
memcached.sess_persistent => 0 => 0
memcached.sess_prefix => memc.sess. => memc.sess.
memcached.sess_randomize_replica_read => 0 => 0
memcached.sess_remove_failed_servers => 0 => 0
memcached.sess_sasl_password => no value => no value
memcached.sess_sasl_username => no value => no value
memcached.sess_server_failure_limit => 0 => 0
memcached.store_retry_count => 2 => 2
```

The script I make to install everything from scratch is here: https://github.com/jonathanpmartins/ubuntusetup/blob/master/server/setup-php7-nginx-1.9.10-ubuntu-14.04.sh. Ignore the ```apt-get install php-memcached``` command on the shell, I suppress it!

**Comment by jonathanpmartins** [comment]
@isotopp @mkoppanen @krakjoe 
I forgot to mention.

    version 3.0.0b1

**Comment by mkoppanen** [comment]
@jonathanpmartins,

when you do 'make test' do you have memcached running on 127.0.0.1:11211?

**Comment by oerdnj** [comment]
@pjebs If you are using my ppa, why you just don't `apt-get install php-memcached`?

**Comment by jonathanpmartins** [comment]
@mkoppanen Sorry for come nearly 1 month later, but my jobs makes me fly.

Lets begin:

- First I create a new instance (in my case, aws) with a fresh default **Ubuntu Server 14.04 LTS**
- After Log-in, I follow this commands: (The script will take a wile to complete) 

```shell
ubuntu@host:~$ wget https://raw.githubusercontent.com/jonathanpmartins/ubuntusetup/master/server/setup-php7-nginx-1.9.10-ubuntu-14.04-without-memcached.sh;
ubuntu@host:~$ sudo chmod +x setup-php7-nginx-1.9.10-ubuntu-14.04-without-memcached.sh;
ubuntu@host:~$ sudo ./setup-php7-nginx-1.9.10-ubuntu-14.04-without-memcached.sh;
```

- After the first round I proceed to the memcached preparation, download, compilation and installation.

```shell
ubuntu@host:~$ which phpize;
/usr/bin/phpize
ubuntu@host:~$ which php
/usr/bin/php
ubuntu@host:~$ php -v
PHP 7.0.4-3+deb.sury.org~trusty+4 (cli) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies
```

- First I will install the latest version of libmemcached, that I find here: https://launchpad.net/libmemcached/+download

```shell
ubuntu@host:~$ wget https://launchpad.net/libmemcached/1.0/1.0.18/+download/libmemcached-1.0.18.tar.gz
ubuntu@host:~$ tar -zxvf libmemcached-1.0.18.tar.gz
ubuntu@host:~$ cd libmemcached-1.0.18/
ubuntu@host:~/libmemcached-1.0.18$ ./configure
ubuntu@host:~/libmemcached-1.0.18$ make; # (will take a while)
ubuntu@host:~/libmemcached-1.0.18$ sudo make install;
ubuntu@host:~/libmemcached-1.0.18$ cd ..
ubuntu@host:~$ rm libmemcached-1.0.18.tar.gz
ubuntu@host:~$ rm -rf libmemcached-1.0.18/
```

- At this point, everything seem to work, I continue:

```shell
ubuntu@host:~$ git clone git@github.com:php-memcached-dev/php-memcached.git
ubuntu@host:~$ cd php-memcached/
ubuntu@host:~/php-memcached$ git checkout php7
```

```shell
ubuntu@host:~/php-memcached$ phpize
Configuring for:
PHP Api Version:         20151012
Zend Module Api No:      20151012
Zend Extension Api No:   320151012
```

```shell
ubuntu@host:~/php-memcached$ ./configure
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for PHP prefix... /usr
checking for PHP includes... -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib
checking for PHP extension directory... /usr/lib/php/20151012
checking for PHP installed headers prefix... /usr/include/php/20151012
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... no
configure: WARNING: You will need re2c 0.13.4 or later if you want to regenerate PHP parsers.
checking for gawk... gawk
checking whether to enable memcached support... yes, shared
checking for libmemcached... yes, shared
checking whether to enable memcached session handler support... yes
checking whether to enable memcached igbinary serializer support... no
checking whether to enable memcached json serializer support... no
checking whether to enable memcached msgpack serializer support... no
checking whether to enable memcached sasl support... yes
checking whether to enable memcached protocol support... no
checking whether to use system FastLZ bibrary... no
checking for ZLIB... yes, shared
checking for pkg-config... /usr/bin/pkg-config
checking for zlib location... /usr
checking for session includes... /usr/include/php/20151012
checking for memcached session support... enabled
checking for memcached igbinary support... disabled
checking for memcached msgpack support... disabled
checking for libmemcached location... found version 1.0.18, under /usr/local
checking whether to enable sasl support... yes
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking sasl/sasl.h usability... no
checking sasl/sasl.h presence... no
checking for sasl/sasl.h... no
configure: error: no, sasl.h is not available. Run configure with --disable-memcached-sasl to disable this check
ubuntu@ip-172-31-47-127:~/php-memcached$ ./configure --disable-memcached-sasl
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for PHP prefix... /usr
checking for PHP includes... -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib
checking for PHP extension directory... /usr/lib/php/20151012
checking for PHP installed headers prefix... /usr/include/php/20151012
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... no
configure: WARNING: You will need re2c 0.13.4 or later if you want to regenerate PHP parsers.
checking for gawk... gawk
checking whether to enable memcached support... yes, shared
checking for libmemcached... yes, shared
checking whether to enable memcached session handler support... yes
checking whether to enable memcached igbinary serializer support... no
checking whether to enable memcached json serializer support... no
checking whether to enable memcached msgpack serializer support... no
checking whether to enable memcached sasl support... no
checking whether to enable memcached protocol support... no
checking whether to use system FastLZ bibrary... no
checking for ZLIB... yes, shared
checking for pkg-config... /usr/bin/pkg-config
checking for zlib location... /usr
checking for session includes... /usr/include/php/20151012
checking for memcached session support... enabled
checking for memcached igbinary support... disabled
checking for memcached msgpack support... disabled
checking for libmemcached location... found version 1.0.18, under /usr/local
checking whether to enable sasl support... no
checking whether memcached_exist is defined... yes
checking for memcached protocol support... disabled
checking how to print strings... printf
checking for a sed that does not truncate output... (cached) /bin/sed
checking for fgrep... /bin/grep -F
checking for ld used by cc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking for gawk... (cached) gawk
checking command to parse /usr/bin/nm -B output from cc object... ok
checking for sysroot... no
checking for mt... mt
checking if mt is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if cc supports -fno-rtti -fno-exceptions... no
checking for cc option to produce PIC... -fPIC -DPIC
checking if cc PIC flag -fPIC -DPIC works... yes
checking if cc static flag -static works... yes
checking if cc supports -c -o file.o... yes
checking if cc supports -c -o file.o... (cached) yes
checking whether the cc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
configure: creating ./config.status
config.status: creating config.h
config.status: executing libtool commands
```

```shell
ubuntu@host:~/php-memcached$ make
/bin/bash /home/ubuntu/php-memcached/libtool --mode=compile cc -I/usr/include/php/20151012    -I. -I/home/ubuntu/php-memcached -DPHP_ATOM_INC -I/home/ubuntu/php-memcached/include -I/home/ubuntu/php-memcached/main -I/home/ubuntu/php-memcached -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include  -DHAVE_CONFIG_H  -g -O2 -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include   -c /home/ubuntu/php-memcached/php_memcached.c -o php_memcached.lo 
libtool: compile:  cc -I/usr/include/php/20151012 -I. -I/home/ubuntu/php-memcached -DPHP_ATOM_INC -I/home/ubuntu/php-memcached/include -I/home/ubuntu/php-memcached/main -I/home/ubuntu/php-memcached -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include -DHAVE_CONFIG_H -g -O2 -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include -c /home/ubuntu/php-memcached/php_memcached.c  -fPIC -DPIC -o .libs/php_memcached.o
/home/ubuntu/php-memcached/php_memcached.c: In function 's_stat_execute_cb':
/home/ubuntu/php-memcached/php_memcached.c:2652:3: warning: passing argument 4 of 'add_assoc_stringl_ex' discards 'const' qualifier from pointer target type [enabled by default]
   add_assoc_stringl_ex(server_values, key, key_length, value, value_length);
   ^
In file included from /usr/include/php/20151012/main/php.h:40:0,
                 from /home/ubuntu/php-memcached/php_memcached.h:22,
                 from /home/ubuntu/php-memcached/php_memcached.c:25:
/usr/include/php/20151012/Zend/zend_API.h:397:14: note: expected 'char *' but argument is of type 'const char *'
 ZEND_API int add_assoc_stringl_ex(zval *arg, const char *key, size_t key_len, char *str, size_t length);
              ^
/home/ubuntu/php-memcached/php_memcached.c: In function 's_server_cursor_list_servers_cb':
/home/ubuntu/php-memcached/php_memcached.c:3370:2: warning: passing argument 4 of 'add_assoc_string_ex' discards 'const' qualifier from pointer target type [enabled by default]
  add_assoc_string(&array, "host", memcached_server_name(instance));
  ^
In file included from /usr/include/php/20151012/main/php.h:40:0,
                 from /home/ubuntu/php-memcached/php_memcached.h:22,
                 from /home/ubuntu/php-memcached/php_memcached.c:25:
/usr/include/php/20151012/Zend/zend_API.h:396:14: note: expected 'char *' but argument is of type 'const char *'
 ZEND_API int add_assoc_string_ex(zval *arg, const char *key, size_t key_len, char *str);
              ^
/home/ubuntu/php-memcached/php_memcached.c:3372:2: warning: passing argument 4 of 'add_assoc_string_ex' discards 'const' qualifier from pointer target type [enabled by default]
  add_assoc_string(&array, "type", memcached_server_type(instance));
  ^
In file included from /usr/include/php/20151012/main/php.h:40:0,
                 from /home/ubuntu/php-memcached/php_memcached.h:22,
                 from /home/ubuntu/php-memcached/php_memcached.c:25:
/usr/include/php/20151012/Zend/zend_API.h:396:14: note: expected 'char *' but argument is of type 'const char *'
 ZEND_API int add_assoc_string_ex(zval *arg, const char *key, size_t key_len, char *str);
              ^
/bin/bash /home/ubuntu/php-memcached/libtool --mode=compile cc -I/usr/include/php/20151012    -I. -I/home/ubuntu/php-memcached -DPHP_ATOM_INC -I/home/ubuntu/php-memcached/include -I/home/ubuntu/php-memcached/main -I/home/ubuntu/php-memcached -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include  -DHAVE_CONFIG_H  -g -O2 -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include   -c /home/ubuntu/php-memcached/php_libmemcached_compat.c -o php_libmemcached_compat.lo 
libtool: compile:  cc -I/usr/include/php/20151012 -I. -I/home/ubuntu/php-memcached -DPHP_ATOM_INC -I/home/ubuntu/php-memcached/include -I/home/ubuntu/php-memcached/main -I/home/ubuntu/php-memcached -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include -DHAVE_CONFIG_H -g -O2 -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include -c /home/ubuntu/php-memcached/php_libmemcached_compat.c  -fPIC -DPIC -o .libs/php_libmemcached_compat.o
/bin/bash /home/ubuntu/php-memcached/libtool --mode=compile cc -I/usr/include/php/20151012    -I. -I/home/ubuntu/php-memcached -DPHP_ATOM_INC -I/home/ubuntu/php-memcached/include -I/home/ubuntu/php-memcached/main -I/home/ubuntu/php-memcached -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include  -DHAVE_CONFIG_H  -g -O2 -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include   -c /home/ubuntu/php-memcached/g_fmt.c -o g_fmt.lo 
libtool: compile:  cc -I/usr/include/php/20151012 -I. -I/home/ubuntu/php-memcached -DPHP_ATOM_INC -I/home/ubuntu/php-memcached/include -I/home/ubuntu/php-memcached/main -I/home/ubuntu/php-memcached -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include -DHAVE_CONFIG_H -g -O2 -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include -c /home/ubuntu/php-memcached/g_fmt.c  -fPIC -DPIC -o .libs/g_fmt.o
/bin/bash /home/ubuntu/php-memcached/libtool --mode=compile cc -I/usr/include/php/20151012    -I. -I/home/ubuntu/php-memcached -DPHP_ATOM_INC -I/home/ubuntu/php-memcached/include -I/home/ubuntu/php-memcached/main -I/home/ubuntu/php-memcached -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include  -DHAVE_CONFIG_H  -g -O2 -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include   -c /home/ubuntu/php-memcached/fastlz/fastlz.c -o fastlz/fastlz.lo 
libtool: compile:  cc -I/usr/include/php/20151012 -I. -I/home/ubuntu/php-memcached -DPHP_ATOM_INC -I/home/ubuntu/php-memcached/include -I/home/ubuntu/php-memcached/main -I/home/ubuntu/php-memcached -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include -DHAVE_CONFIG_H -g -O2 -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include -c /home/ubuntu/php-memcached/fastlz/fastlz.c  -fPIC -DPIC -o fastlz/.libs/fastlz.o
/bin/bash /home/ubuntu/php-memcached/libtool --mode=compile cc -I/usr/include/php/20151012    -I. -I/home/ubuntu/php-memcached -DPHP_ATOM_INC -I/home/ubuntu/php-memcached/include -I/home/ubuntu/php-memcached/main -I/home/ubuntu/php-memcached -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include  -DHAVE_CONFIG_H  -g -O2 -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include   -c /home/ubuntu/php-memcached/php_memcached_session.c -o php_memcached_session.lo 
libtool: compile:  cc -I/usr/include/php/20151012 -I. -I/home/ubuntu/php-memcached -DPHP_ATOM_INC -I/home/ubuntu/php-memcached/include -I/home/ubuntu/php-memcached/main -I/home/ubuntu/php-memcached -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include -DHAVE_CONFIG_H -g -O2 -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include -c /home/ubuntu/php-memcached/php_memcached_session.c  -fPIC -DPIC -o .libs/php_memcached_session.o
/bin/bash /home/ubuntu/php-memcached/libtool --mode=link cc -DPHP_ATOM_INC -I/home/ubuntu/php-memcached/include -I/home/ubuntu/php-memcached/main -I/home/ubuntu/php-memcached -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include  -DHAVE_CONFIG_H  -g -O2 -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include   -o memcached.la -export-dynamic -avoid-version -prefer-pic -module -rpath /home/ubuntu/php-memcached/modules  php_memcached.lo php_libmemcached_compat.lo g_fmt.lo fastlz/fastlz.lo php_memcached_session.lo -Wl,-rpath,/usr/local/lib -L/usr/local/lib -lz -lmemcached -lmemcachedutil
libtool: link: cc -shared  -fPIC -DPIC  .libs/php_memcached.o .libs/php_libmemcached_compat.o .libs/g_fmt.o fastlz/.libs/fastlz.o .libs/php_memcached_session.o   -L/usr/local/lib -lz /usr/local/lib/libmemcached.so /usr/local/lib/libmemcachedutil.so  -O2 -Wl,-rpath -Wl,/usr/local/lib   -pthread -Wl,-soname -Wl,memcached.so -o .libs/memcached.so
libtool: link: ( cd ".libs" && rm -f "memcached.la" && ln -s "../memcached.la" "memcached.la" )
/bin/bash /home/ubuntu/php-memcached/libtool --mode=install cp ./memcached.la /home/ubuntu/php-memcached/modules
libtool: install: cp ./.libs/memcached.so /home/ubuntu/php-memcached/modules/memcached.so
libtool: install: cp ./.libs/memcached.lai /home/ubuntu/php-memcached/modules/memcached.la
libtool: finish: PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/sbin" ldconfig -n /home/ubuntu/php-memcached/modules
----------------------------------------------------------------------
Libraries have been installed in:
   /home/ubuntu/php-memcached/modules

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the `-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
     during execution
   - add LIBDIR to the `LD_RUN_PATH' environment variable
     during linking
   - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
   - have your system administrator add LIBDIR to `/etc/ld.so.conf'

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------

Build complete.
Don't forget to run 'make test'.
```

```shell
ubuntu@host:~/php-memcached$ make test

Build complete.
Don't forget to run 'make test'.


=====================================================================
PHP         : /usr/bin/php7.0 
PHP_SAPI    : cli
PHP_VERSION : 7.0.4-3+deb.sury.org~trusty+4
ZEND_VERSION: 3.0.0
PHP_OS      : Linux - Linux ip-172-31-47-127 3.13.0-74-generic #118-Ubuntu SMP Thu Dec 17 22:52:10 UTC 2015 x86_64
INI actual  : /home/ubuntu/php-memcached/tmp-php.ini
More .INIs  :   
CWD         : /home/ubuntu/php-memcached
Extra dirs  : 
VALGRIND    : Not used
=====================================================================
TIME START 2016-03-08 13:27:22
=====================================================================
PASS Check for memcached presence [tests/001.phpt] 
FAIL Memcached::add() [tests/add.phpt] 
FAIL Memcached::append() [tests/append.phpt] 
PASS Memcached construct with bad arguments [tests/bad_construct.phpt] 
PASS Memcached: Bug #16084 (Crash when addServers is called with an associative array) [tests/bug_16084.phpt] 
FAIL Memcached: Bug #16959 (getMulti + BINARY_PROTOCOL problem) [tests/bug_16959.phpt] 
FAIL Change prefix, pecl bug #17137 [tests/bug_17137.phpt] 
FAIL Memcached::getServerByKey(): Bug pecl#18639 (Segfault in getServerByKey) [tests/bug_18639.phpt] 
FAIL Memcached::get() with cache callback [tests/cachecallback.phpt] 
PASS make sure that callback exception behaves correctly [tests/callback_exception.phpt] 
PASS Callback initializer throws and dies [tests/callback_exception_2.phpt] 
FAIL Memcached fetch cas & set cas [tests/cas.phpt] 
FAIL Memcached multi fetch cas & set cas [tests/cas_multi.phpt] 
PASS Check if persistent object is persistent [tests/check_if_persistent.phpt] 
PASS Check if persistent object is new or an old persistent one [tests/check_if_pristine.phpt] 
PASS Test cloning [tests/clone.phpt] 
FAIL Memcached compression test [tests/compression_types.phpt] 
FAIL Conf settings persist. [tests/conf_persist.phpt] 
PASS Memcached constructor [tests/construct.phpt] 
PASS persistent memcached connection [tests/construct_persistent.phpt] 
PASS Default behaviors [tests/default_behavior.phpt] 
PASS Memcached store & fetch type correctness [tests/deleted.phpt] 
FAIL Delete multi [tests/deletemulti.phpt] 
PASS Delete multi key types [tests/deletemultitypes.phpt] 
FAIL Memcached::addByKey() [tests/experimental/add_bykey.phpt] 
PASS Memcached::addServer() unix doamin socket [tests/experimental/addserver_unixdomain.phpt] 
FAIL Memcached::appendByKey() [tests/experimental/append_bykey.phpt] 
FAIL Memcached::casByKey() [tests/experimental/cas_bykey.phpt] 
FAIL Memcached::cas() with strange key [tests/experimental/cas_invalid_key.phpt] 
FAIL Memcached::deleteByKey() [tests/experimental/delete_bykey.phpt] 
FAIL Delete multi with integer keys [tests/experimental/deletemulti_nonstringkeys.phpt] 
FAIL Extreme floats: max, min, Inf, -Inf, and NaN [tests/experimental/extreme_floats.phpt] 
FAIL Memcached getDelayed() and fetch() with and without cas [tests/experimental/fetch.phpt] 
FAIL Memcached::fetch() with bad unserialize [tests/experimental/fetch_badunserialize.phpt] 
FAIL Memcached::fetch() with bad unserialize [tests/experimental/fetchall_badunserialize.phpt] 
FAIL Memcached::get() [tests/experimental/get.phpt] 
FAIL Memcached::getByKey() [tests/experimental/get_bykey.phpt] 
FAIL Memcached::getByKey() with CAS [tests/experimental/get_bykey_cas.phpt] 
FAIL Memcached::set()/delete() UDP [tests/experimental/get_udp.phpt] 
PASS Memcached::getDelayedByKey() with bad server [tests/experimental/getdelayed_badserver.phpt] 
FAIL Memcached::getDelayed() with bad unserialize [tests/experimental/getdelayed_badunserialize.phpt] 
FAIL Memcached::getDelayedByKey() [tests/experimental/getdelayed_bykey.phpt] 
FAIL Memcached::getDelayedByKey() with CAS [tests/experimental/getdelayed_bykey_cas.phpt] 
FAIL Memcached::getDelayedByKey() with callback exception [tests/experimental/getdelayed_cbthrows.phpt] 
FAIL Memcached getDelayed non string keys [tests/experimental/getdelayed_nonstring_keys.phpt] 
FAIL Memcached::getMulti() bad server [tests/experimental/getmulti_badserver.phpt] 
FAIL Memcached::getMulti() with bad unserialize [tests/experimental/getmulti_badunserialize.phpt] 
FAIL Memcached::getMultiByKey() [tests/experimental/getmulti_bykey.phpt] 
FAIL Memcached::getMulti() with empty array [tests/experimental/getmulti_empty.phpt] 
FAIL Memcached::getMulti() partial error [tests/experimental/getmulti_partial_error.phpt] 
FAIL Memcached::getVersion() [tests/experimental/getversion.phpt] 
FAIL Float should not consider locale [tests/experimental/locale_float.phpt] 
PHP Warning:  preg_match(): Compilation failed: missing ) at offset 35 in /home/ubuntu/php-memcached/run-tests.php on line 2237
PHP Warning:  preg_match(): Compilation failed: unmatched parentheses at offset 2 in /home/ubuntu/php-memcached/run-tests.php on line 2237
FAIL Memcached::phpinfo() [tests/experimental/moduleinfo.phpt] 
FAIL Memcached::appendByKey() [tests/experimental/prepend_bykey.phpt] 
FAIL Memcached::replaceByKey() [tests/experimental/replace_bykey.phpt] 
FAIL Serializer basic [tests/experimental/serializer/serializer_php.phpt] 
SKIP Serializer: exception while serializing [tests/experimental/serializer/serializer_php_bad_serialize.phpt] reason: skip when using PHP internal
FAIL Serializer: exception while unserializing [tests/experimental/serializer/serializer_php_bad_unserialize.phpt] 
SKIP Serialize igbinary [tests/experimental/serializer_igbinary.phpt] reason: no igbinary loadedskip igbinary support not enabled
SKIP Serialize JSON [tests/experimental/serializer_json.phpt] reason: no JSON loadedskip JSON support not enabled
FAIL Session expiration [tests/experimental/session_gc.phpt] 
FAIL Memcached::setByKey() [tests/experimental/set_bykey.phpt] 
FAIL Compress below factor and fail to plain. [tests/experimental/set_comp_below_factor.phpt] 
PASS Set default serializer [tests/experimental/set_default_serializer.phpt] 
PASS Set invalid serializer [tests/experimental/set_invalid_serializer.phpt] 
FAIL Compress with 0 factor and get [tests/experimental/setget_zero_factor.phpt] 
FAIL Memcached::setMultiByKey() with bad serialize [tests/experimental/setmulti_badserialize.phpt] 
FAIL Memcached::setMultiByKey() [tests/experimental/setmulti_bykey.phpt] 
FAIL Memcached::getStats() [tests/experimental/stats.phpt] 
FAIL Memcached::getStats() with bad server [tests/experimental/stats_badserver.phpt] 
XFAIL Memcached store, fetch & touch expired key [tests/expire.phpt]   XFAIL REASON: https://code.google.com/p/memcached/issues/detail?id=275
FAIL Test flushing buffers [tests/flush_buffers.phpt] 
FAIL Memcached::get/getMulti() flags [tests/get_flags.phpt] 
FAIL Memcached getDelayed callback [tests/getdelayed.phpt] 
FAIL Memcached::getMulti() [tests/getmulti.phpt] 
PASS Memcached::getServerByKey() [tests/getserverbykey.phpt] 
PASS getServerList [tests/getserverlist.phpt] 
FAIL Test for bug 155 [tests/gh_155.phpt] 
FAIL Test for Github issue 21 [tests/gh_21.phpt] 
FAIL Test for Github issue #77 [tests/gh_77.phpt] 
FAIL Test for GH #90 [tests/gh_90.phpt] 
FAIL Test for Github issue #93 (double and long overflow) [tests/gh_93.phpt] 
FAIL Memcached::increment() Memcached::decrement() [tests/incrdecr.phpt] 
FAIL Memcached::incrementByKey() Memcached::decrementByKey() [tests/incrdecr_bykey.phpt] 
FAIL Memcached::increment() Memcached::decrement() with initial support [tests/incrdecr_initial.phpt] 
PASS Memcached::increment() Memcached::decrement() with invalid key [tests/incrdecr_invalid_key.phpt] 
PASS Get version [tests/invalid_options.phpt] 
PASS Test that callback is invoked on new object [tests/invoke_callback.phpt] 
PASS Use callback initializer [tests/invoke_callback_2.phpt] 
PASS Test that callback is invoked on new object only once [tests/invoke_callback_twice.phpt] 
FAIL Test different kind of keys [tests/keys.phpt] 
FAIL Memcached local server test [tests/localserver.phpt] 
FAIL Memcached GET_PRESERVE_ORDER flag in getMulti [tests/multi_order.phpt] 
PASS Test that correct return value is returned [tests/no-not-found.phpt] 
PASS Memcached options [tests/options.phpt] 
FAIL Wrong return values for binary protocol [tests/pr_75.phpt] 
FAIL Memcached::prepend() [tests/prepend.phpt] 
FAIL Memcached::replace() [tests/replace.phpt] 
FAIL Memcached result codes. [tests/rescode.phpt] 
SKIP Test SASL authentication [tests/sasl_basic.phpt] reason: no SASL support enabled
FAIL Session bad configurations, prefix [tests/session_badconf_emptyprefix.phpt] 
SKIP Session bad configurations, lock wait time [tests/session_badconf_locktime.phpt] reason: skip until timeout in memcached set
PASS Session bad configurations, prefix [tests/session_badconf_prefix.phpt] 
PASS Session bad configurations, invalid save path (server list) [tests/session_badconf_servers.phpt] 
FAIL Session basic open, write, destroy [tests/session_basic.phpt] 
FAIL Session basic open, write, destroy [tests/session_basic2.phpt] 
FAIL Session basic open, write, destroy [tests/session_basic3.phpt] 
FAIL Session lock [tests/session_lock.phpt] 
FAIL Session regenerate [tests/session_regenerate.phpt] 
FAIL set large data [tests/set_large.phpt] 
FAIL Memcached::setMulti() [tests/setmulti.phpt] 
PASS Set options using setOptions [tests/setoptions.phpt] 
FAIL Check stats [tests/stats.phpt] 
FAIL Touch in binary mode [tests/touch_binary.phpt] 
SKIP Memcached store & fetch type and value correctness using igbinary serializer [tests/types_igbinary.phpt] reason: igbinary not enabled
SKIP Memcached multi store & multi fetch type and value correctness using igbinary serializer [tests/types_igbinary_multi.phpt] reason: igbinary not enabled
SKIP Memcached store & fetch type and value correctness using JSON serializer [tests/types_json.phpt] reason: json not enabled
SKIP Memcached multi store & multi fetch type and value correctness using JSON serializer [tests/types_json_multi.phpt] reason: json not enabled
SKIP Memcached store & fetch type and value correctness using msgpack serializer [tests/types_msgpack.phpt] reason: msgpack not enabled
SKIP Memcached multi store & fetch type and value correctness using msgpack serializer [tests/types_msgpack_multi.phpt] reason: msgpack not enabled
FAIL Memcached store & fetch type and value correctness using PHP serializer [tests/types_php.phpt] 
FAIL Memcached multi store & fetch type and value correctness using PHP serializer [tests/types_php_multi.phpt] 
FAIL Set with undefined key and value [tests/undefined_set.phpt] 
FAIL Memcached user flags [tests/user-flags.phpt] 
PASS Memcached virtual buckets [tests/vbucket.phpt] 
FAIL Get version [tests/version.phpt] 
=====================================================================
TIME END 2016-03-08 13:28:27

=====================================================================
TEST RESULT SUMMARY
---------------------------------------------------------------------
Exts skipped    :    0
Exts tested     :   14
---------------------------------------------------------------------

Number of tests :  126               115
Tests skipped   :   11 (  8.7%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :   84 ( 66.7%) ( 73.0%)
Expected fail   :    1 (  0.8%) (  0.9%)
Tests passed    :   30 ( 23.8%) ( 26.1%)
---------------------------------------------------------------------
Time taken      :   65 seconds
=====================================================================

=====================================================================
EXPECTED FAILED TEST SUMMARY
---------------------------------------------------------------------
Memcached store, fetch & touch expired key [tests/expire.phpt]  XFAIL REASON: https://code.google.com/p/memcached/issues/detail?id=275
=====================================================================

=====================================================================
FAILED TEST SUMMARY
---------------------------------------------------------------------
Memcached::add() [tests/add.phpt]
Memcached::append() [tests/append.phpt]
Memcached: Bug #16959 (getMulti + BINARY_PROTOCOL problem) [tests/bug_16959.phpt]
Change prefix, pecl bug #17137 [tests/bug_17137.phpt]
Memcached::getServerByKey(): Bug pecl#18639 (Segfault in getServerByKey) [tests/bug_18639.phpt]
Memcached::get() with cache callback [tests/cachecallback.phpt]
Memcached fetch cas & set cas [tests/cas.phpt]
Memcached multi fetch cas & set cas [tests/cas_multi.phpt]
Memcached compression test [tests/compression_types.phpt]
Conf settings persist. [tests/conf_persist.phpt]
Delete multi [tests/deletemulti.phpt]
Memcached::addByKey() [tests/experimental/add_bykey.phpt]
Memcached::appendByKey() [tests/experimental/append_bykey.phpt]
Memcached::casByKey() [tests/experimental/cas_bykey.phpt]
Memcached::cas() with strange key [tests/experimental/cas_invalid_key.phpt]
Memcached::deleteByKey() [tests/experimental/delete_bykey.phpt]
Delete multi with integer keys [tests/experimental/deletemulti_nonstringkeys.phpt]
Extreme floats: max, min, Inf, -Inf, and NaN [tests/experimental/extreme_floats.phpt]
Memcached getDelayed() and fetch() with and without cas [tests/experimental/fetch.phpt]
Memcached::fetch() with bad unserialize [tests/experimental/fetch_badunserialize.phpt]
Memcached::fetch() with bad unserialize [tests/experimental/fetchall_badunserialize.phpt]
Memcached::get() [tests/experimental/get.phpt]
Memcached::getByKey() [tests/experimental/get_bykey.phpt]
Memcached::getByKey() with CAS [tests/experimental/get_bykey_cas.phpt]
Memcached::set()/delete() UDP [tests/experimental/get_udp.phpt]
Memcached::getDelayed() with bad unserialize [tests/experimental/getdelayed_badunserialize.phpt]
Memcached::getDelayedByKey() [tests/experimental/getdelayed_bykey.phpt]
Memcached::getDelayedByKey() with CAS [tests/experimental/getdelayed_bykey_cas.phpt]
Memcached::getDelayedByKey() with callback exception [tests/experimental/getdelayed_cbthrows.phpt]
Memcached getDelayed non string keys [tests/experimental/getdelayed_nonstring_keys.phpt]
Memcached::getMulti() bad server [tests/experimental/getmulti_badserver.phpt]
Memcached::getMulti() with bad unserialize [tests/experimental/getmulti_badunserialize.phpt]
Memcached::getMultiByKey() [tests/experimental/getmulti_bykey.phpt]
Memcached::getMulti() with empty array [tests/experimental/getmulti_empty.phpt]
Memcached::getMulti() partial error [tests/experimental/getmulti_partial_error.phpt]
Memcached::getVersion() [tests/experimental/getversion.phpt]
Float should not consider locale [tests/experimental/locale_float.phpt]
Memcached::phpinfo() [tests/experimental/moduleinfo.phpt]
Memcached::appendByKey() [tests/experimental/prepend_bykey.phpt]
Memcached::replaceByKey() [tests/experimental/replace_bykey.phpt]
Serializer basic [tests/experimental/serializer/serializer_php.phpt]
Serializer: exception while unserializing [tests/experimental/serializer/serializer_php_bad_unserialize.phpt]
Session expiration [tests/experimental/session_gc.phpt]
Memcached::setByKey() [tests/experimental/set_bykey.phpt]
Compress below factor and fail to plain. [tests/experimental/set_comp_below_factor.phpt]
Compress with 0 factor and get [tests/experimental/setget_zero_factor.phpt]
Memcached::setMultiByKey() with bad serialize [tests/experimental/setmulti_badserialize.phpt]
Memcached::setMultiByKey() [tests/experimental/setmulti_bykey.phpt]
Memcached::getStats() [tests/experimental/stats.phpt]
Memcached::getStats() with bad server [tests/experimental/stats_badserver.phpt]
Test flushing buffers [tests/flush_buffers.phpt]
Memcached::get/getMulti() flags [tests/get_flags.phpt]
Memcached getDelayed callback [tests/getdelayed.phpt]
Memcached::getMulti() [tests/getmulti.phpt]
Test for bug 155 [tests/gh_155.phpt]
Test for Github issue 21 [tests/gh_21.phpt]
Test for Github issue #77 [tests/gh_77.phpt]
Test for GH #90 [tests/gh_90.phpt]
Test for Github issue #93 (double and long overflow) [tests/gh_93.phpt]
Memcached::increment() Memcached::decrement() [tests/incrdecr.phpt]
Memcached::incrementByKey() Memcached::decrementByKey() [tests/incrdecr_bykey.phpt]
Memcached::increment() Memcached::decrement() with initial support [tests/incrdecr_initial.phpt]
Test different kind of keys [tests/keys.phpt]
Memcached local server test [tests/localserver.phpt]
Memcached GET_PRESERVE_ORDER flag in getMulti [tests/multi_order.phpt]
Wrong return values for binary protocol [tests/pr_75.phpt]
Memcached::prepend() [tests/prepend.phpt]
Memcached::replace() [tests/replace.phpt]
Memcached result codes. [tests/rescode.phpt]
Session bad configurations, prefix [tests/session_badconf_emptyprefix.phpt]
Session basic open, write, destroy [tests/session_basic.phpt]
Session basic open, write, destroy [tests/session_basic2.phpt]
Session basic open, write, destroy [tests/session_basic3.phpt]
Session lock [tests/session_lock.phpt]
Session regenerate [tests/session_regenerate.phpt]
set large data [tests/set_large.phpt]
Memcached::setMulti() [tests/setmulti.phpt]
Check stats [tests/stats.phpt]
Touch in binary mode [tests/touch_binary.phpt]
Memcached store & fetch type and value correctness using PHP serializer [tests/types_php.phpt]
Memcached multi store & fetch type and value correctness using PHP serializer [tests/types_php_multi.phpt]
Set with undefined key and value [tests/undefined_set.phpt]
Memcached user flags [tests/user-flags.phpt]
Get version [tests/version.phpt]
=====================================================================

You may have found a problem in PHP.
This report can be automatically sent to the PHP QA team at
http://qa.php.net/reports and http://news.php.net/php.qa.reports
This gives us a better understanding of PHP's behavior.
If you don't want to send the report immediately you can choose
option "s" to save it.	You can then email it to qa-reports@lists.php.net later.
Do you want to send this report now? [Yns]: Y

Please enter your email address.
(Your address will be mangled so that it will not go out on any
mailinglist in plain text): web@jonathanmartins.com.br

Posting to http://qa.php.net/buildtest-process.php

Thank you for helping to make PHP better.
```

... and here is where I get stuck. Am I forgoting something? ```igbinary``` is required?

**Comment by jonathanpmartins** [comment]
@mkoppanen Sorry for come nearly 1 month later, but my jobs makes me fly.

I get It work, Lets begin:

- First I create a new instance (in my case, aws) with a fresh default **Ubuntu Server 14.04 LTS**
- After Log-in, I follow this commands: (The script will take a wile to complete) 

```shell
ubuntu@host:~$ wget https://raw.githubusercontent.com/jonathanpmartins/ubuntusetup/master/server/setup-php7-nginx-1.9.10-ubuntu-14.04-without-memcached.sh;
ubuntu@host:~$ sudo chmod +x setup-php7-nginx-1.9.10-ubuntu-14.04-without-memcached.sh;
ubuntu@host:~$ sudo ./setup-php7-nginx-1.9.10-ubuntu-14.04-without-memcached.sh;
```

- After the first round I proceed to the memcached preparation, download, compilation and installation.

```shell
ubuntu@host:~$ which phpize;
/usr/bin/phpize
ubuntu@host:~$ which php
/usr/bin/php
ubuntu@host:~$ php -v
PHP 7.0.4-3+deb.sury.org~trusty+4 (cli) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies
```

- First I will install the latest version of libmemcached, that I find here: https://launchpad.net/libmemcached/+download

```shell
ubuntu@host:~$ wget https://launchpad.net/libmemcached/1.0/1.0.18/+download/libmemcached-1.0.18.tar.gz
ubuntu@host:~$ tar -zxvf libmemcached-1.0.18.tar.gz
ubuntu@host:~$ cd libmemcached-1.0.18/
ubuntu@host:~/libmemcached-1.0.18$ ./configure
ubuntu@host:~/libmemcached-1.0.18$ make; # (will take a while)
ubuntu@host:~/libmemcached-1.0.18$ sudo make install;
ubuntu@host:~/libmemcached-1.0.18$ cd ..
ubuntu@host:~$ rm libmemcached-1.0.18.tar.gz
ubuntu@host:~$ rm -rf libmemcached-1.0.18/
```

- At this point, everything seem to work, I continue:

```shell
ubuntu@host:~$ git clone git@github.com:php-memcached-dev/php-memcached.git
ubuntu@host:~$ cd php-memcached/
ubuntu@host:~/php-memcached$ git checkout php7
```

```shell
ubuntu@host:~/php-memcached$ phpize
Configuring for:
PHP Api Version:         20151012
Zend Module Api No:      20151012
Zend Extension Api No:   320151012
```

```shell
ubuntu@host:~/php-memcached$ ./configure
```

```shell
ubuntu@host:~/php-memcached$ make
----------------------------------------------------------------------
Libraries have been installed in:
   /home/ubuntu/php-memcached/modules

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the `-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
     during execution
   - add LIBDIR to the `LD_RUN_PATH' environment variable
     during linking
   - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
   - have your system administrator add LIBDIR to `/etc/ld.so.conf'

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------

Build complete.
Don't forget to run 'make test'.
```

```shell
ubuntu@host:~/php-memcached$ make test
/bin/bash /home/ubuntu/php-memcached/libtool --mode=install cp ./memcached.la /home/ubuntu/php-memcached/modules
libtool: install: cp ./.libs/memcached.so /home/ubuntu/php-memcached/modules/memcached.so
libtool: install: cp ./.libs/memcached.lai /home/ubuntu/php-memcached/modules/memcached.la
libtool: finish: PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/sbin" ldconfig -n /home/ubuntu/php-memcached/modules
----------------------------------------------------------------------
Libraries have been installed in:
   /home/ubuntu/php-memcached/modules

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the `-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
     during execution
   - add LIBDIR to the `LD_RUN_PATH' environment variable
     during linking
   - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
   - have your system administrator add LIBDIR to `/etc/ld.so.conf'

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------

Build complete.
Don't forget to run 'make test'.


=====================================================================
PHP         : /usr/bin/php7.0 
PHP_SAPI    : cli
PHP_VERSION : 7.0.4-3+deb.sury.org~trusty+4
ZEND_VERSION: 3.0.0
PHP_OS      : Linux - Linux ip-172-31-47-127 3.13.0-74-generic #118-Ubuntu SMP Thu Dec 17 22:52:10 UTC 2015 x86_64
INI actual  : /home/ubuntu/php-memcached/tmp-php.ini
More .INIs  :   
CWD         : /home/ubuntu/php-memcached
Extra dirs  : 
VALGRIND    : Not used
=====================================================================
TEST RESULT SUMMARY
---------------------------------------------------------------------
Exts skipped    :    0
Exts tested     :   14
---------------------------------------------------------------------

Number of tests :  126               115
Tests skipped   :   11 (  8.7%) --------
Tests warned    :    1 (  0.8%) (  0.9%)
Tests failed    :   17 ( 13.5%) ( 14.8%)
Expected fail   :    0 (  0.0%) (  0.0%)
Tests passed    :   97 ( 77.0%) ( 84.3%)
---------------------------------------------------------------------
Time taken      :   18 seconds
=====================================================================

=====================================================================
FAILED TEST SUMMARY
---------------------------------------------------------------------
Memcached::casByKey() [tests/experimental/cas_bykey.phpt]
Memcached::cas() with strange key [tests/experimental/cas_invalid_key.phpt]
Memcached::deleteByKey() [tests/experimental/delete_bykey.phpt]
Memcached getDelayed() and fetch() with and without cas [tests/experimental/fetch.phpt]
Memcached::get() [tests/experimental/get.phpt]
Memcached::getByKey() with CAS [tests/experimental/get_bykey_cas.phpt]
Memcached::getDelayedByKey() [tests/experimental/getdelayed_bykey.phpt]
Memcached::getDelayedByKey() with CAS [tests/experimental/getdelayed_bykey_cas.phpt]
Memcached getDelayed non string keys [tests/experimental/getdelayed_nonstring_keys.phpt]
Memcached::getMulti() bad server [tests/experimental/getmulti_badserver.phpt]
Memcached::getMultiByKey() [tests/experimental/getmulti_bykey.phpt]
Memcached::getMulti() with empty array [tests/experimental/getmulti_empty.phpt]
Memcached::phpinfo() [tests/experimental/moduleinfo.phpt]
Compress with 0 factor and get [tests/experimental/setget_zero_factor.phpt]
Memcached::setMultiByKey() with bad serialize [tests/experimental/setmulti_badserialize.phpt]
Memcached::getStats() [tests/experimental/stats.phpt]
Memcached::getStats() with bad server [tests/experimental/stats_badserver.phpt]
=====================================================================

=====================================================================
WARNED TEST SUMMARY
---------------------------------------------------------------------
Memcached store, fetch & touch expired key [tests/expire.phpt] (warn: XFAIL section but test passes)
=====================================================================

You may have found a problem in PHP.
This report can be automatically sent to the PHP QA team at
http://qa.php.net/reports and http://news.php.net/php.qa.reports
This gives us a better understanding of PHP's behavior.
If you don't want to send the report immediately you can choose
option "s" to save it.	You can then email it to qa-reports@lists.php.net later.
Do you want to send this report now? [Yns]: Y

Please enter your email address.
(Your address will be mangled so that it will not go out on any
mailinglist in plain text): web@jonathanmartins.com.br

Posting to http://qa.php.net/buildtest-process.php

Thank you for helping to make PHP better.
```

... and here is where I get stuck. But when I see you

Trying to install it, even with tests failing:
```shell
ubuntu@host:~/php-memcached$ sudo make install
Installing shared extensions:     /usr/lib/php/20151012/
```

Installation seems to work, now I can see this information on php info:

```shell
ubuntu@host:~/php-memcached$ php -i | grep memcached
/etc/php/7.0/cli/conf.d/20-memcached.ini,
memcached
memcached support => enabled
libmemcached version => 1.0.18
memcached.compression_factor => 1.3 => 1.3
memcached.compression_threshold => 2000 => 2000
memcached.compression_type => fastlz => fastlz
memcached.default_binary_protocol => 0 => 0
memcached.default_connect_timeout => 0 => 0
memcached.default_consistent_hash => 0 => 0
memcached.serializer => php => php
memcached.sess_binary_protocol => 1 => 1
memcached.sess_connect_timeout => 0 => 0
memcached.sess_consistent_hash => 1 => 1
memcached.sess_lock_expire => 0 => 0
memcached.sess_lock_max_wait => not set => not set
memcached.sess_lock_retries => 5 => 5
memcached.sess_lock_wait => not set => not set
memcached.sess_lock_wait_max => 2000 => 2000
memcached.sess_lock_wait_min => 1000 => 1000
memcached.sess_locking => 1 => 1
memcached.sess_number_of_replicas => 0 => 0
memcached.sess_persistent => 0 => 0
memcached.sess_prefix => memc.sess. => memc.sess.
memcached.sess_randomize_replica_read => 0 => 0
memcached.sess_remove_failed_servers => 0 => 0
memcached.sess_sasl_password => no value => no value
memcached.sess_sasl_username => no value => no value
memcached.sess_server_failure_limit => 0 => 0
memcached.store_retry_count => 2 => 2
```
PHPINFO
```
memcached support => enabled
Version => 3.0.0b1
libmemcached version => 1.0.18
SASL support => no
Session support => yes
igbinary support => no
json support => no
msgpack support => no
```

And thats when I got it to work fine!

**Comment by krakjoe** [comment]
I'm going to close this bug.

The original issue was a misconfiguration one, not a bug.

The failing tests you are seeing are in the experimental folder, the README is not very precise about how to run tests.

If you want to run tests like travis does, do ```make test TESTS=tests/*.phpt", this will skip experimental tests.

If you think you have an issue, please open a new ticket.

**Comment by emiglobetrotting** [comment]
php-memcached test all skipped with one common reason "cannot connect to server"

I'm trying to install it on Centos 7 and these are my steps

    yum -y install memcached libmemcached-devel
    
made changes in nano /etc/sysconfig/memcached to fit the options below

    PORT=”1337″
    USER=”memcached”
    MAXCONN=”2000″
    CACHESIZE=”4096″
    OPTIONS=”-l 127.0.0.1”

    systemctl restart memcached 

    systemctl start memcached

    systemctl enable memcached

    cd php-src/ext

    git clone https://github.com/php-memcached-dev/php-memcached.git

    cd php-memcached

    /usr/local/php7/bin/phpize

    ./configure --prefix='/usr/local/php7' --with-libmemcached-dir='/usr' --disable-memcached-sasl --with-php-config='/usr/local/php7/bin/php-config'


    make && make install

    make test //all test result is skipped
